viva_las_vegas_script_tests.py

```diff
@@ -462,6 +462,133 @@ def test_sending_request_and_rejecting_it_without_permission(self):
+        )
+        assert self.check_latest_response_with_retries(
+            'Тогда сначала предупреди, а потом вернись и повтори все снова!'
+        )
+
+    def test_sending_ill_request_with_working_from_home_and_rejecting_it(self):
+        """Tests if it's possible to send an ill request with working home. """
+
+        self.choose_general_channel()
+
+        self.send_message(
+            '{} болею'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Очень жаль. Ты в состоянии работать из дома в эти дни?\n'
+            'Да\n'
+            'Нет'
+        )
+
+        self.send_message(
+            '{} Болею и работаю'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Я понял. Согласовано ли отсутствие с руководителем/тимлидом?\n'
+            'Да\n'
+            'Нет'
+        )
+
+        self.send_message(
+            '{} Нет, они не предупреждены, что я болею'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Тогда сначала предупреди, а потом вернись и повтори все снова!'
+        )
+
+    def test_sending_ill_request(self):
+        """Tests if it's possible to send an ill and approve it. """
+
+        self.choose_general_channel()
+
+        self.send_message(
+            '{} болею'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Очень жаль. Ты в состоянии работать из дома в эти дни?\n'
+            'Да\n'
+            'Нет'
+        )
+
+        self.send_message(
+            '{} Болею и работаю'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Я понял. Согласовано ли отсутствие с руководителем/тимлидом?\n'
+            'Да\n'
+            'Нет'
+        )
+
+        self.send_message(
+            '{} Да, они предупреждены, что я болею'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Ok. Выздоравливай поскорее. Я добавил событие в календарь. '
+            'Когда ты выйдешь на работу, скажи мне `я не болею`.'
+        )
+
+        self.switch_channel('leave-coordination')
+        assert self.check_latest_response_with_retries(
+            '@{} болеет и работает из дома'.format(self.username)
+        )
+
+    def test_sending_ill_request_when_previous_one_is_approved(self):
+        """Tests if it's not possible to send an ill when you already have another one. """
+
+        self.choose_general_channel()
+
+        self.send_message(
+            '{} болею'.format(self._bot_name)
+        )
+        assert self.check_latest_response_with_retries(
+            'Я уже слышал, что ты болеешь. 🤔'
+        )
+
+    def test_cancelling_approved_ill_request(self):
+        """Tests if it's possible to cancel an ill request. """
```

Comment:
Tests if it's possible to cancel the ill request which has already been approved.